### PR TITLE
Making stunnel request/limit configurable

### PR DESCRIPTION
**Pull request by roopakparikh**
Making the stunnel request/limit configurable.
Use of env varilable STUNNEL_REQUEST_CPU, STUNNEL_REQUEST_MEMORY,
STUNNEL_LIMIT_CPU, STUNNEL_LIMIT_MEMORY all will change the
configuration on all the stunnel containers. If the request cpu &
memory are missing default value of 10Mi (memory) and 5m (cpu) would
be used, if the LIMIT values are not specified limit would be skipped

### CODE REVIEW

File: pkg/k8sutil/stunnel.go [hunk 1]
```diff
@@ -114,6 +115,32 @@ func InsertStunnel(
...
 	})
+
+
+	// Default values for stunnel request and memory
+	// stunnel has been observed to consume between 3 and 6 MB, so
+	// set memory request to 5 and limit to 25.
+	// Also limit cpu usage to 1 whole CPU
+	// In some cases we have seen the memory limit to go higher > 50Mi
+	// these cases can be handled by the configuration variables
+	var limitCPU string = getEnvVarWithDefault("STUNNEL_LIMIT_CPU", "")
```
Comment (bcle) [inline pkg/k8sutil/stunnel.go]: more idiomatic: `limitCPU := getEnvVarWithDefault("STUNNEL_LIMIT_CPU", "")`
same comment for next 2 lines

File: pkg/k8sutil/stunnel.go [hunk 2]
```diff
@@ -130,21 +157,17 @@ func InsertStunnel(
...
 	return volumes, containers
 }
 
 
+func getEnvVarWithDefault(key string, defaultVal string) string{
+	val := os.Getenv(key)
+	if val == "" {
+		return defaultVal
+	}
+       return val
```
Comment (bcle) [inline pkg/k8sutil/stunnel.go]: remove extra space

File: pkg/k8sutil/stunnel.go (hunk 1 above)
Comment (roopakparikh) [reply pkg/k8sutil/stunnel.go]: ok, refreshing

File: pkg/k8sutil/stunnel.go (hunk 2 above)
Comment (roopakparikh) [reply pkg/k8sutil/stunnel.go]: ok, refreshing

File: pkg/k8sutil/stunnel.go (hunk 1 above)
Comment (roopakparikh) [reply pkg/k8sutil/stunnel.go]: Updated

File: pkg/k8sutil/stunnel.go (hunk 2 above)
Comment (roopakparikh) [reply pkg/k8sutil/stunnel.go]: Updated

**Comment by roopakparikh** [comment]
@bcle fixed, please review and let me know if you want me to merge it.

File: pkg/k8sutil/stunnel.go [hunk 3]
```diff
@@ -21,48 +22,48 @@ func stunnelEnvVars(
 	index int,
 ) []v1.EnvVar {
 
-	springboardListenPort := fmt.Sprintf("%d", 6789 + index)
-	stunnelEnv := []v1.EnvVar {
+	springboardListenPort := fmt.Sprintf("%d", 6789+index)
+	stunnelEnv := []v1.EnvVar{
 		{
-			Name: "STUNNEL_VERIFY_CHAIN",
+			Name:  "STUNNEL_VERIFY_CHAIN",
```
Comment (bcle) [inline pkg/k8sutil/stunnel.go]: why was a space inserted?

File: pkg/k8sutil/stunnel.go [hunk 4]
```diff
@@ -2,7 +2,8 @@ package k8sutil
 
 import (
 	"fmt"
-	"k8s.io/api/core/v1"
+	"os"
+        "k8s.io/api/core/v1"
```
Comment (bcle) [inline pkg/k8sutil/stunnel.go]: remove extra space

File: pkg/k8sutil/stunnel.go (hunk 3 above)
Comment (roopakparikh) [reply pkg/k8sutil/stunnel.go]: These formatting changes are thanks to VS Code, not sure why it is doing that.

File: pkg/k8sutil/stunnel.go (hunk 4 above)
Comment (roopakparikh) [reply pkg/k8sutil/stunnel.go]: Let me clean up, it is something to do with my VS code formatting, I will remove it, this is annoying.

**Comment by roopakparikh** [comment]
@bcle I have refreshed it again, manually applying the patch

File: pkg/k8sutil/stunnel.go [hunk 5]
```diff
@@ -2,6 +2,7 @@ package k8sutil
 
 import (
 	"fmt"
+        "os"
```
Comment (bcle) [inline pkg/k8sutil/stunnel.go]: remove space

File: pkg/k8sutil/stunnel.go [hunk 6]
```diff
@@ -114,7 +115,30 @@ func InsertStunnel(
 			},
 		},
 	})
-	containers = append(containers, v1.Container{
+	// Default values for stunnel request and memory
```
Comment (bcle) [inline pkg/k8sutil/stunnel.go]: Not sure why, the indentation of these 2 lines still looks incorrect

**Pull request merged**